extensions/interactions/GraphInput/directives/graph-viz.component.spec.ts

```diff
@@ -0,0 +1,1881 @@
+      component.state.hoveredVertex = 0;
+      component.state.addEdgeVertex = 0;
+      spyOn(component, 'tryAddEdge').and.callThrough();
+
+      expect(component.isMobile).toBe(false);
+      expect(component.graph.edges).toEqual([
+        {
+          src: 0,
+          weight: 1,
+          dst: 1
+        },
+        {
+          src: 1,
+          weight: 1,
+          dst: 2
+        }
+      ]);
+
+      component.onMouseupDocument();
+      tick(10);
+
+      expect(component.tryAddEdge).toHaveBeenCalledWith(0, 0);
+      expect(component.graph.edges).toEqual([
+        {
+          src: 0,
+          weight: 1,
+          dst: 1
+        },
+        {
+          src: 1,
+          weight: 1,
+          dst: 2
+        }
+      ]);
+    }));
+
+  it('should not add edge if an edge an edge is already present between' +
+  'the destination and source for a non directed graph', fakeAsync(() => {
+    component.state.currentMode = component._MODES.ADD_EDGE;
+    component.state.hoveredVertex = 0;
+    component.state.addEdgeVertex = 1;
+    spyOn(component, 'tryAddEdge').and.callThrough();
+    spyOn(component, 'endAddEdge').and.callThrough();
+
+    expect(component.isMobile).toBe(false);
+    expect(component.graph.edges).toEqual([
+      {
+        src: 0,
+        weight: 1,
+        dst: 1
+      },
+      {
+        src: 1,
+        weight: 1,
+        dst: 2
+      }
+    ]);
+
+    component.onMouseupDocument();
+    tick(10);
+
+    expect(component.tryAddEdge).toHaveBeenCalledWith(1, 0);
+    expect(component.graph.edges).toEqual([
+      {
+        src: 0,
+        weight: 1,
+        dst: 1
+      },
+      {
+        src: 1,
+        weight: 1,
+        dst: 2
+      }
+    ]);
+    expect(component.endAddEdge).toHaveBeenCalled();
+  }));
+
+  it('should end add edge when user clicks anything other than a' +
+  ' vertex', fakeAsync(() => {
+    component.state.currentMode = component._MODES.ADD_EDGE;
+    component.state.addEdgeVertex = 0;
+    spyOn(component, 'endAddEdge').and.callThrough();
+
+    component.onMouseupDocument();
+    tick(10);
+
+    expect(component.state.hoveredVertex).toBeNull();
+    expect(component.endAddEdge).toHaveBeenCalled();
+    expect(component.state.addEdgeVertex).toBeNull();
+  }));
+
+  it('should end moving vertex when user releases mouse button',
+    fakeAsync(() => {
+      component.state.currentMode = component._MODES.MOVE;
+      component.state.currentlyDraggedVertex = 0;
+      spyOn(component, 'endDragVertex').and.callThrough();
+
+      component.onMouseupDocument();
+      tick(10);
+
+      expect(component.endDragVertex).toHaveBeenCalled();
+      expect(component.state.currentlyDraggedVertex).toBeNull();
+    }));
+
+  it('should execute debounce when user releases mouse button', () => {
+    // Currently there is no method to test a private Method Decorator.
+    // The following lines execute but DO NOT test the debounce private
```

Comment:
Did you check [this](https://netbasal.com/create-and-test-decorators-in-javascript-85e8d5cf879c) out? It seems to explain how we can test decorators in jest, the method should be the same for jasmine. Can you confirm if this way works?

Reply:
Done, Thanks for the help!